osbenchmark/aggregator.py

```diff
@@ -13,23 +13,31 @@ def __init__(self, cfg, test_executions_dict, args):
         self.args = args
         self.test_executions = test_executions_dict
         self.accumulated_results: Dict[str, Dict[str, List[Any]]] = {}
-        self.accumulated_iterations: Dict[str, int] = {}
+        self.accumulated_iterations: Dict[str, Dict[str, int]] = {}
         self.metrics = ["throughput", "latency", "service_time", "client_processing_time", "processing_time", "error_rate", "duration"]
         self.test_store = metrics.test_execution_store(self.config)
         self.cwd = cfg.opts("node", "benchmark.cwd")
 
-    def count_iterations_for_each_op(self) -> None:
+    def count_iterations_for_each_op(self, test_execution) -> None:
         loaded_workload = workload.load_workload(self.config)
```

Comment:
Seeing if we can improve this method: 
- Notice that we are loading the workload each time we pass in a test execution. Could we load the workload once and just reuse it?
- See that this method has some steps that checks if the test procedure is valid. Might be better to move that to the designated method `test_execution_compatibility_check`

Reply:
To your first point: Good call, this should be notably faster for larger aggregations now that I just load the workload once.

Also removed that extra logic👍